components/eam/src/chemistry/modal_aero/modal_aero_drydep.F90

```diff
@@ -0,0 +1,736 @@
+
+    ! constants
+
+    real(r8) gamma(11)      ! exponent of schmidt number
+    data gamma/0.56e+00_r8,  0.54e+00_r8,  0.54e+00_r8,  0.56e+00_r8,  0.56e+00_r8, &        
+               0.56e+00_r8,  0.50e+00_r8,  0.54e+00_r8,  0.54e+00_r8,  0.54e+00_r8, &
+               0.54e+00_r8/
+    save gamma
+
+    real(r8) alpha(11)      ! parameter for impaction
+    data alpha/1.50e+00_r8,   1.20e+00_r8,  1.20e+00_r8,  0.80e+00_r8,  1.00e+00_r8, &
+               0.80e+00_r8, 100.00e+00_r8, 50.00e+00_r8,  2.00e+00_r8,  1.20e+00_r8, &
+              50.00e+00_r8/
+    save alpha
+
+    real(r8) radius_collector(11) ! radius (m) of surface collectors
+    data radius_collector/10.00e-03_r8,  3.50e-03_r8,  3.50e-03_r8,  5.10e-03_r8,  2.00e-03_r8, &
+                           5.00e-03_r8, -1.00e+00_r8, -1.00e+00_r8, 10.00e-03_r8,  3.50e-03_r8, &
+                          -1.00e+00_r8/
+    save radius_collector
+
+    integer  :: iwet(11) ! flag for wet surface = 1, otherwise = -1
+    data iwet/-1,  -1,   -1,   -1,   -1,  &
+              -1,   1,   -1,    1,   -1,  &
+              -1/
+    save iwet
+
+
+    ! use a maximum radius of 50 microns when calculating deposition velocity
+
+    real(r8),parameter :: radiaus_max = 50.0e-6_r8
+
+    !------------------------------------------------------------------------
+    ! Calculate particle velocity of gravitational settling
+    !------------------------------------------------------------------------
+    do kk=1,pver
+       do ii=1,ncol
+
+          lnsig = log(sig_part(ii,kk))
+          radius_moment(ii,kk) = min(radiaus_max,radius_part(ii,kk))*exp((float(moment)-1.5_r8)*lnsig*lnsig)
+          dispersion = exp(2._r8*lnsig*lnsig)
+
+          rho=pmid(ii,kk)/rair/tair(ii,kk)
+
+          ! Quasi-laminar layer resistance: call rss_lmn_get
+          ! Size-independent thermokinetic properties
+
+          vsc_dyn_atm(ii,kk) = 1.72e-5_r8 * ( (tair(ii,kk)/273.0_r8)**1.5_r8) * 393.0_r8 &
+                                             /(tair(ii,kk)+120.0_r8)  ![kg m-1 s-1] RoY94 p. 102
+          mfp_atm(ii,kk) = 2.0_r8 * vsc_dyn_atm(ii,kk) / &   ![m] SeP97 p. 455
+                          ( pmid(ii,kk)*sqrt(8.0_r8/(pi*rair*tair(ii,kk))) )
+          vsc_knm_atm(ii,kk) = vsc_dyn_atm(ii,kk) / rho ![m2 s-1] Kinematic viscosity of air
+
+          slp_crc(ii,kk) = 1.0_r8 + mfp_atm(ii,kk) * &
+                           (1.257_r8+0.4_r8*exp(-1.1_r8*radius_moment(ii,kk)/(mfp_atm(ii,kk)))) / &
+                           radius_moment(ii,kk)   ![frc] Slip correction factor SeP97 p. 464
+
+          vlc_grv(ii,kk) = (4.0_r8/18.0_r8) * radius_moment(ii,kk)*radius_moment(ii,kk)*density_part(ii,kk)* &
+                           gravit*slp_crc(ii,kk) / vsc_dyn_atm(ii,kk) ![m s-1] Stokes' settling velocity SeP97 p. 466
+          vlc_grv(ii,kk) = vlc_grv(ii,kk) * dispersion
+
+       enddo
+    enddo
+    vlc_dry(:ncol,:)=vlc_grv(:ncol,:)
```

Comment:
We might further make two subroutines, one for the above calculation and one for the below calculation.